test/module/irohad/ametsuchi/ametsuchi_test.cpp

```diff
@@ -866,3 +867,30 @@ TEST_F(AmetsuchiTest, FindTxByHashTest) {
   ASSERT_EQ(*blocks->getTxByHashSync(tx2hash), tx2);
   ASSERT_EQ(blocks->getTxByHashSync(tx3hash), boost::none);
 }
+
+/**
+ * @given initialized storage for ordering service
+ * @when save proposal height
+ * @then load proposal height and ensure it is correct
+ */
+TEST_F(AmetsuchiTest, OrderingServicePersistentStorageTest) {
+  std::shared_ptr<iroha::ametsuchi::PostgresOrderingServicePersistentState> ordering_state;
+  auto orderingStorageResult = iroha::ametsuchi::PostgresOrderingServicePersistentState::create(pgopt_);
+  orderingStorageResult.match(
+      [&](iroha::expected::Value<std::shared_ptr<iroha::ametsuchi::PostgresOrderingServicePersistentState>> &_storage) {
+        ordering_state = _storage.value;
+      },
+      [](iroha::expected::Error<std::string> &error) {
+        FAIL() << "PostgresOrderingServicePersistentState: " << error.error;
+      });
+  ASSERT_TRUE(ordering_state);
+
+  ordering_state->reset();
+  ASSERT_EQ(2, ordering_state->loadProposalHeight().value());
+  ASSERT_TRUE(ordering_state->saveProposalHeight(11));
+  ASSERT_EQ(11, ordering_state->loadProposalHeight().value());
+  ASSERT_TRUE(ordering_state->saveProposalHeight(33));
+  ASSERT_EQ(33, ordering_state->loadProposalHeight().value());
+  ordering_state->reset();
+  ASSERT_EQ(2, ordering_state->loadProposalHeight().value());
+}
```

Comment:
Need to add tests where:
* same as first but connections are different
* simulate a shutdown of storage and check the state after restoration.

Reply:
what do you mean by different connections?

Reply:
I mean different connections means different objects of ordering_state and different physical connections.